frontend/test/metabase/scenarios/visualizations/pie_chart.cy.spec.js

```diff
@@ -0,0 +1,52 @@
+import { restore, visitQuestionAdhoc } from "__support__/cypress";
+import { SAMPLE_DATASET } from "__support__/cypress_sample_dataset";
+
+const { PRODUCTS, PRODUCTS_ID } = SAMPLE_DATASET;
+
+const testQuery = {
+  type: "query",
+  query: {
+    "source-table": PRODUCTS_ID,
+    aggregation: [["count"]],
+    breakout: [["field-id", PRODUCTS.CATEGORY]],
+  },
+  database: 1,
+};
+
+describe("scenarios > visualizations > pie chart", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  function ensurePieChartRendered(rows, totalValue) {
+    cy.get(".Visualization").within(() => {
+      // detail
+      cy.findByText("Total").should("be.visible");
+      cy.findByTestId("detail-value").should("have.text", totalValue);
+
+      // slices
+      cy.findAllByTestId("slice").should("have.length", rows.length);
+
+      // legend
+      rows.forEach((name, i) => {
+        cy.get(".LegendItem")
+          .contains(name)
+          .should("exist");
+      });
+    });
+  }
+
+  it("renders a pie chart", () => {
+    cy.server();
```

Comment:
I bit of background behind using `cy.server()` and `cy.route()`.

Cypress moved to `cy.intercept()` which removes the need to start `cy.server()` but we can't currently upgrade to the v6+ and are stuck with the old pattern.

The best place to put `cy.server()` is in the top-most `beforeEach()` hook. I wanted to go through ALL tests and apply that change but there's always some higher priority task at hand.

So, when creating new files, let's keep it in the beforeEach.